core/unix/os.c

```diff
@@ -2012,7 +2012,7 @@ os_tls_thread_exit(local_state_t *local_state)
 
     /* We already set TLS to &uninit_tls in os_thread_exit() */
 
-    if (dynamo_exited && !last_thread_tls_exited) {
+    if (!client_thread && dynamo_exited && !last_thread_tls_exited) {
```

Comment:
Won't we still have a problem if client threads are present and a non-client thread calls dr_exit_process()?  last_thread_tls_exited will be set too early then?